velox/functions/prestosql/aggregates/MapAggAggregate.cpp

```diff
@@ -81,7 +81,12 @@ class MapAggAggregate : public exec::Aggregate {
       offset += mapSize;
     }
 
-    *result = removeDuplicates(std::dynamic_pointer_cast<MapVector>(*result));
+    // canonicalize requires a singly referenced MapVector. std::move
```

Comment:
This pattern seems fragile. There is no guarantee that "result" is singly-referenced, is there?

Reply:
Lost comment?

---
This pattern seems fragile. There is no guarantee that "result" is singly-referenced, is there?